src/espic/solve_maxwell.py

```diff
@@ -34,3 +35,103 @@ def solve(self, rho):
         phi[1:-1] = solve_banded((1, 1), bands, rhs)
 
         return phi
+
+
+# Code taken from https://john-s-butler-dit.github.io/NumericalAnalysisBook/Chapter%2009%20-%20Elliptic%20Equations/903_Poisson%20Equation-Boundary.html
+# FIXME: for now, this assumes equal spacing in x and y.
+class MaxwellSolver2D:
```

Comment:
Is there a way to make MaxwellSolver2D handle the 1D case easily? This would make the code simpler and easier to test and use. (I'm thinking of NumPy-style vectorization.)

Reply:
Not easily. The matrix in 1D is strictly tri-diagonal, whereas it's only block tri-diagonal in 2D. This affects the linear solve, since solving a tri-diagonal system is very fast, but less fast otherwise. I think it's okay to separate the two versions if one of them can use more specialized solvers.

A different possible fix would be having a generic MaxwellSolver class, which can switch to 1D or 2D.

Reply:
It's probably not impossible, but maybe not super important right now. I'd have to think of a slick way of doing it that doesn't make the code less readable.